### PR TITLE
feat: first audit changes

### PR DESCRIPTION
**Pull request by DinshawKothari**
Ticket: WIN-00000

### CODE REVIEW

File: contracts/USDS.sol [hunk 1]
```diff
@@ -100,8 +101,38 @@ contract USDS is
     function setProofOfReserveFeed(
         address newFeedAddress
     ) external onlyRole(DEFAULT_ADMIN_ROLE) {
-        require(newFeedAddress != address(0), "Cannot add zero address as a proof of reserve feed");
-        proofOfReserveFeed = AggregatorV3Interface(newFeedAddress);
+        require(
```
Comment (sachushaji) [inline contracts/USDS.sol:104]: You can utilise validateProofOfReserve.  You can still make use of the above change of directly setting proofOfReserveFeed and if the feed is stale validateProofOfReserve should get things reverted. Please write a test case for that

File: contracts/USDS.sol [hunk 2]
```diff
@@ -100,8 +101,38 @@ contract USDS is
...
+            "Cannot add zero address as a proof of reserve feed"
+        );
+
+        // ERC165 check to ensure the new feed implements AggregatorV3Interface
+        require(
+            IERC165(newFeedAddress).supportsInterface(
+                type(AggregatorV3Interface).interfaceId
+            ),
+            "New feed must implement AggregatorV3Interface"
+        );
```
Comment (sachushaji) [inline contracts/USDS.sol:115]: We need to check whether chainlink actually supports this

File: contracts/USDS.sol [hunk 3]
```diff
@@ -126,9 +157,16 @@ contract USDS is
...
-        require(newAddress != address(0), "Cannot add zero address as a reserve address");
-        reserveAddresses[newAddress] = true;
-        emit ReserveAddressAdded(newAddress);
+        require(
+            newAddress != address(0),
+            "Cannot add zero address as a reserve address"
+        );
+
+        // Check if the address is already a reserve address
+        if (!reserveAddresses[newAddress]) {
```
Comment (sachushaji) [inline contracts/USDS.sol:166]: May be we can revert this if the reserver address is already set instead of this if condition

**Comment by DinshawKothari** [comment]
As discussed made changes in the new PR - https://github.com/BitGo/USDS/pull/33

**Pull request closed without merging**